### DISCUSSION TITLE
Feature Request: Provide ARM64 binaries

### DISCUSSION THREAD

**Discussion by Risae** [rabbitmq/erlang-rpm#108] [General]
Hi RabbitMQ Community!

It would be super awesome if you could provide ARM64 compiled binaries, so we can utitilize the power of ARM CPUs.

Thank you!

**Comment by michaelklishin** [comment]
Our team is small and we are not in the business of packaging Erlang.
So perhaps someone else in the community should take the charge on this initiative, like those who need latest Erlang RPMs on ARM64.

We have invested a lot of time into x86-64 and Docker image/OCI packaging.
ARM64 packages of Erlang are already available [for Debian](https://launchpad.net/~rabbitmq/+archive/ubuntu/rabbitmq-erlang) and for [OCI users](https://github.com/docker-library/rabbitmq/) with complete automation.

Those looking to contribute can probably rely on [Fedora Copr](https://copr.fedorainfracloud.org/)
and adapt this repo or at least the patches in it to match the expectations of Copr.

Another aspect of any additional packaging work that you ask us to take is who is going to foot the
bill for the bandwidth. Erlang RPM and Debian packages for x86-64 have a footprint of a few TBs a month. No package hosting provider would provide that much bandwidth for free. Maybe this is also where Fedora Corp would help, I haven't looked closely.

**Comment by michaelklishin** [comment]
@Risae would you be interested in contributing in ARM64 builds of this package via Copr or any other way? I'm afraid our team does not have the resources to provide ARM64 packages. So the users who need
those packages should scratch their own itch instead of waiting for us.

**Reply by Risae** [reply]
> @Risae would you be interested in contributing in ARM64 builds of this package via Copr or any other way? I'm afraid our team does not have the resources to provide ARM64 packages. So the users who need those packages should scratch their own itch instead of waiting for us.

Hi  michaelklishin!

Thank you very much for replying so quickly. I'm sadly not a developer, just a regular RabbitMQ maintainer and i don't have the necessary skills to create a pipeline to build ARM64 binaries. ARM64 in general is a lot less power hungry and more efficient compared to regular x86, and this is one of the reasons why it would be great if you could provide such an option for environment conscious users like myself. I know it would be a lot of work to create such a pipeline, but i believe that ARM64 CPUs will get even more popular and the decision not to provide such a binary might hinder the success of RabbitMQ in the future. I hope you guys can reconsider this decision! But aside from that i want to thank you so much for providing Zero-dependency Erlang RPM for RabbitMQ in the first place, you guys rock!!

**Reply by michaelklishin** [reply]
@Risae no need to tout the benefits of ARM, we understand them well.

Our team has already contributed [to] automation of latest Erlang releases on ARM64 for Debian ([via Launchpad](https://launchpad.net/~rabbitmq/+archive/ubuntu/rabbitmq-erlang)) and the [OCI](https://github.com/docker-library/rabbitmq/) image. We do not have the resources to do the same for this RPM, at least for the foreseeable future.

"Regular RabbitMQ users" have been getting this software and our Erlang packages for free for years, and many make a profit from systems built with RabbitMQ and Erlang. At least some should consider contributing, open source software should not be taken for granted.

**Comment by michaelklishin** [comment]
If someone would be interested in adapting this package for Fedora Corp which can produce ARM64 builds, let us know.

All discussions that are not related to potential contributions and just want to put pressure on us to do all the work so that others can get this stuff for free will be deleted.

**Comment by michaelklishin** [comment]
#115 is relevant for this discussion

**Comment by michaelklishin** [comment]
[25.3](https://github.com/rabbitmq/erlang-rpm/releases/tag/v25.3) is the first release with aarch64 packages provided in binary form.

As of #119, anyone with an ARM64 host and Docker installed can build aarm64 versions of this package
for CentOS Stream 9, Stream 8, Fedora 38+ and Amazon Linux 2023.

aarch64 packages are not built by automatic pipelines, so I'd have to trigger their builds on my ARM64 host every so often (perhaps not for every patch release).

Most importantly, #119 proves that the package did not have any aarch64 incompatibilities, it was a matter of modernizing the spec a bit and automating Docker builds for non-CentOS Stream distributions.
